### PR TITLE
add doc for sql_select_limit

### PR DESCRIPTION
**Pull request by hanfei1991**
Signed-off-by: Han Fei <hanfei19910905@gmail.com>

### What is changed, added or deleted? (Required)

Support mysql system variable `sql_select_limit`

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/3515
- Other reference link(s): https://github.com/pingcap/tidb/pull/17604

### CODE REVIEW

**Comment by hanfei1991** [comment]
@zz-jason @SunRunAway  PTAL

**Comment by hanfei1991** [comment]
@TomShawn PTAL

File: system-variables.md [hunk 1]
```diff
@@ -118,6 +118,7 @@ The following MySQL system variables are fully supported in TiDB and have the sa
 | max\_execution\_time | GLOBAL \| SESSION | the execution timeout for a statement, in milliseconds |
 | innodb\_lock\_wait\_timeout | GLOBAL \| SESSION | the lock wait time for pessimistic transactions, in seconds |
 | windowing\_use\_high\_precision | GLOBAL \| SESSION |  whether to use the high precision mode when computing the window function. The default value is ON |
+| sql\_select\_limit | GLOBAL \| SESSION | The maximum number of rows to return from SELECT statements. The default value is 2^64 - 1. |
```
Comment (TomShawn) [inline system-variables.md:121]:
```suggestion
| sql\_select\_limit | GLOBAL \| SESSION | The maximum number of rows to return from the `SELECT` statements. The default value is `2^64 - 1`. |
```

**Pull request merged**